src/app/public/modules/text-editor/services/text-editor-adapter.service.ts

```diff
@@ -0,0 +1,532 @@
+
+    documentEl.addEventListener('selectionchange', selectionListener);
+    documentEl.addEventListener('input', selectionListener);
+    documentEl.addEventListener('mousedown', clickListener);
+    documentEl.body.addEventListener('paste', pasteListener);
+    return {
+      iframeElementRef: element,
+      selectionListener: selectionListener,
+      clickListener: clickListener,
+      pasteListener: pasteListener,
+      selectionChangeObservable: selectionObservable,
+      clickObservable: clickObservable,
+      commandChangeObservable: new Subject()
+    };
+  }
+
+  public static removeObservers(setting: EditorSetting): void {
+    /* istanbul ignore next */
+    const documentEl = setting.iframeElementRef.contentWindow ?
+       setting.iframeElementRef.contentWindow.document :
+       setting.iframeElementRef.contentDocument;
+    setting.selectionChangeObservable.complete();
+    setting.clickObservable.complete();
+    setting.commandChangeObservable.complete();
+    documentEl.removeEventListener('selectionchange', setting.selectionListener);
+    documentEl.removeEventListener('input', setting.selectionListener);
+    documentEl.removeEventListener('mousedown', setting.clickListener);
+    documentEl.body.removeEventListener('paste', setting.pasteListener);
+  }
+
+  /* istanbul ignore next */
+  private getPasteOverride(id: string): (e: ClipboardEvent) => void {
+    return (e: ClipboardEvent): void => {
+      e.preventDefault();
+      const text = e.clipboardData.getData('text/plain');
+      this.execCommand(id, { command: 'insertHTML', value: text });
+    };
+  }
+
+  private getCurrentSelectionParentElement(id: string): Element {
+    return this.selectionService.getCurrentSelectionParentElement(this.getDocumentEl(id));
+  }
+
+  private static getColor(documentEl: Document, selector: string): string {
+    const commandValue = documentEl.queryCommandValue(selector);
+
+    // Edge is weird and returns numbers
+    if (typeof commandValue === 'number') {
+      /* istanbul ignore next */
+      // tslint:disable-next-line: no-bitwise
+      return 'rgb(' + (commandValue & 0xFF) + ', ' +
+                    // tslint:disable-next-line: no-bitwise
+                    ((commandValue & 0xFF00) >> 8) + ', ' +
+                    // tslint:disable-next-line: no-bitwise
+                    ((commandValue & 0xFF0000) >> 16 ) + ')';
+    }
+
+    // Firefox uses 'Transparent' instead of a color value
+    /* istanbul ignore next */
+    if (commandValue.toString().toLowerCase() === 'transparent') {
+      return STYLE_STATE_DEFAULTS.backColor;
+    }
+    return commandValue;
+  }
+
+  private hasLink(editorId: string): boolean {
+    const anchorEl = this.getSelectedAnchorTag(editorId);
+    const childAnchorEls = this.getChildSelectedAnchorTags(editorId);
+    return childAnchorEls.length > 0 || (!!anchorEl && !!anchorEl.href);
+  }
+
+  private static getParent(element: Element, tag: string): Element {
```

Comment:
If this is private - why is it static?

Reply:
Talked offline - reverted to non-static methods.